pulseapi/profiles/models/categories.py

```diff
@@ -59,3 +63,80 @@ class ProgramYear(models.Model):
 
     def __str__(self):
         return self.value
+
+
+class UserProfileProgramParticipation(models.Model):
+    profile = models.ForeignKey(
+        'profiles.UserProfile',
+        on_delete=models.CASCADE,
+        related_name='program_participation',
+    )
+
+    program = models.ForeignKey(
+        'profiles.ProgramType',
+        on_delete=models.CASCADE,
+        related_name='participation',
+    )
+
+    year = models.PositiveSmallIntegerField(
+        # TODO: Change to MaxValueValidator with callable when
+        #       we update to Django > v2.2
+        validators=[YearValidator(max_offset=2)],
```

Comment:
This feels like something that would need yearly updating... (See comment below)

Reply:
Since the offset is from the current year we won't need to update. I'll peg down the min year (makes sense since it's static). But, for max_year, it will automatically change based on the current year. So every year, when we want to set a value for the `year` property, we can set it up to that year + 2 years ahead. So for example, in 2021, it will automatically allow to set the year to be up to 2023.

Now, the `2` offset is arbitrary (well sorta, I decided based on - we don't plan program members beyond 2 years). Happy to change that to something else.

Reply:
2 sounds reasonable to me.